cmd/wasm/main.go

```diff
@@ -0,0 +1,181 @@
+	Error     string `json:"error,omitempty"`
+	Formatted string `json:"formatted,omitempty"`
+}
+
+// celFmt formats a given string using our CEL (Common Expression Language)
+// formatting rules. It compiles the given program as part of the formatting
+// process. This function takes one argument, which must be a string.
+//
+// The function always returns an object. On success, the object contains one
+// attribute named 'formatted' which contains the formatted CEL program. If any
+// error occurs, then the object contains an attribute named 'error' whose value
+// is the string error message.
+func celFmt(_ js.Value, args []js.Value) any {
+	if len(args) != 1 {
+		return toObject(&celFmtResult{Error: "celFmt requires one argument"})
+	}
+	if args[0].Type() != js.TypeString {
+		return toObject(&celFmtResult{Error: "celFmt argument must be a string"})
+	}
+
+	buf := new(bytes.Buffer)
+	if err := compileAndFormat(buf, args[0].String()); err != nil {
+		return toObject(&celFmtResult{Error: err.Error()})
+	}
+	return toObject(&celFmtResult{Formatted: buf.String()})
+}
+
+// toObject converts a struct to a map[string]any using JSON marshal/unmarshal.
+func toObject(v any) map[string]any {
+	data, err := json.Marshal(v)
+	if err != nil {
+		panic(err)
+	}
+
+	var out map[string]any
+	if err = json.Unmarshal(data, &out); err != nil {
+		panic(err)
+	}
+
+	return out
+}
+
+// moduleBuildMetadata returns a map containing build and dependency metadata
+// about WebAssembly module. This includes version numbers for specific modules
+// and other relevant settings such as commit hash and time.
+//
+// It then returns this map, which may contain the following keys:
+//
+//   - go: The Go language version used to build the program (without the "go" prefix).
+//   - celfmt: The version of the main module in the build information.
+//   - mito: The version of the "github.com/elastic/mito" module.
+//   - cel-go: The version of the "github.com/google/cel-go" module.
+//   - commit: The VCS revision (commit hash) from the build settings, if available.
+//   - commit_time: The timestamp of the VCS revision from the build settings, if available.
+func moduleBuildMetadata(_ js.Value, _ []js.Value) any {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return nil
+	}
+
+	meta := map[string]any{
+		"go":     strings.TrimPrefix(runtime.Version(), "go"),
+		"celfmt": info.Main.Version,
+	}
+
+	for _, m := range info.Deps {
+		switch m.Path {
+		case "github.com/elastic/mito":
+			meta["mito"] = m.Version
+		case "github.com/google/cel-go":
+			meta["cel-go"] = m.Version
+		}
+	}
+
+	for _, setting := range info.Settings {
+		switch setting.Key {
+		case "vcs.revision":
+			meta["commit"] = setting.Value
+		case "vcs.time":
+			meta["commit_time"] = setting.Value
+		}
+	}
+
+	return meta
+}
+
+func main() {
+	done := make(chan int, 0)
+	js.Global().Set("celFmt", js.FuncOf(celFmt))
+	js.Global().Set("celModuleBuildMetadata", js.FuncOf(moduleBuildMetadata))
+	<-done
```

Comment:
```suggestion
	js.Global().Set("celFmt", js.FuncOf(celFmt))
	js.Global().Set("celModuleBuildMetadata", js.FuncOf(moduleBuildMetadata))
	select {}
```
?

Reply:
That makes sense. 

I want to ask @bhapas about whether the module needs to be able to exit. In #3, he has function to close the channel which exits the module. If that's needed then I will need to keep the done channel.

Reply:
If the chan is required, then the len param to make can be dropped.